src/test/java/com/google/step/youtube/DescriptionParserTest.java

```diff
@@ -329,10 +307,149 @@ public void codeWithQuotes_multipleMatches() {
             + "https://darkventures.co.uk/shop with offer code 'bobat15'"
             + "Save 33% on your first Native Deodorant Pack - normally $36, you'll get it for "
             + "$24! Click here: https://bit.ly/nativecoolirpa and use my code \"COOLIRPA\".";
-
         List<String> actual = DescriptionParser.findMatches(CODE_WITH_QUOTES_PATTERN, desc);
         assertThat(actual, equalTo(Arrays.asList("bobat15", "COOLIRPA")));
     }
 
+
+    /*                                                                   *
+     * ============= TEST 'TO' AND 'AT' FOR AFFILATE LINKS ============= *
```

Comment:
I was thinking, as the other regexes are also run in the actual code, is there a guarantee that only to...at would match and produce the correct string rather than a separate regex giving a false result when validating here? You'll likely want to run minimal cases for single regexes and have more cases where you validate a string for all regexes right?

Reply:
The tests for single regexs call findMatches() instead of parse() so it only runs only 1 regex. This way I can guarantee that for all tests below line 316, only TO_AT_LINKS is run. The idea was to test specific edge cases for each individual regex and then test them all generally together (these are the tests at the top of the file).

Reply:
Hmm yeah I'm just wondering if there's some way of organizing the test so it's more readable like input + expected output pairs in lists based on regexes, and then a regex test would just run through those per list, while an overall test would apply all regexes on each entry in all the lists to make sure there's no conflicting overlap...

Kind of like my initial suggestion with having a csv to validate over, but instead of it being in a separate file as discouraged by derrill, just have them be constants defined in the test class. But everything works fine at the moment so it's really just something optional to think about if you want to reuse code more effectively.

Reply:
Ah I see, but would that method make it less apparent which edge case each input/output pair is testing?